core/src/main/java/org/apache/iceberg/rest/RESTFileScanTaskParser.java

```diff
@@ -85,8 +93,8 @@ public static FileScanTask fromJson(
           allDeleteFiles.size());
       deleteFiles =
           indices.stream()
-              .map(index -> (GenericDeleteFile) allDeleteFiles.get(index))
```

Comment:
We don't really need the `GenericDeleteFile` instance, we don't care about how DeleteFiles are handed to us so we should just code to the interface here